core/src/main/java/org/elasticsearch/index/query/ScriptQueryBuilder.java

```diff
@@ -19,24 +19,50 @@
 
 package org.elasticsearch.index.query;
 
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.RandomAccessWeight;
+import org.apache.lucene.search.Weight;
+import org.apache.lucene.util.Bits;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.xcontent.XContentBuilder;
-import org.elasticsearch.script.Script;
+import org.elasticsearch.script.*;
 import org.elasticsearch.script.Script.ScriptField;
+import org.elasticsearch.search.lookup.SearchLookup;
 
 import java.io.IOException;
+import java.util.Objects;
 
 public class ScriptQueryBuilder extends AbstractQueryBuilder<ScriptQueryBuilder> {
 
     public static final String NAME = "script";
 
-    static final ScriptQueryBuilder PROTOTYPE = new ScriptQueryBuilder((Script) null);
+    static final ScriptQueryBuilder PROTOTYPE = new ScriptQueryBuilder();
 
-    private Script script;
+    private final Script script;
+    
+    private ScriptQueryBuilder() {
+        this.script = null;
+    }
 
     public ScriptQueryBuilder(Script script) {
+        if (script == null) {
+            throw new IllegalArgumentException("supplied script cannot be null.");
+        }
```

Comment:
look at BoostingQueryBuilder for instance. I explained in other PR when it makes sense to have checks in setter/constructors rather than validate, this is something that we are adapting as we go. I will say it again: check in setters/constructors are for things that can happen only through java api. depending on the parser, we might check for existence of elements in the parsers, meaning that null values will never come through from the parser to the builder. In that case null checks are only for the java api code and can be moved earlier on. If the check is valid for both rest layer (parser) and java api, then we should have the check in the validate method.